### PR TITLE
Performance monitor

### PR DESCRIPTION
**Pull request by UnrealKaraulov**
I need add performance monitor for test, how to build it with old glibc version?(

### CODE REVIEW

**Comment by UnrealKaraulov** [comment]
How to build it for old glibc version as in website amxmodx.org ?

**Comment by dvander** [comment]
What is going on in this PR? There's no description and there's tons of changes that seem unrelated.

**Comment by UnrealKaraulov** [comment]
@dvander I tried to add performance monitor (If function executed with big time, print it to log)

Changes only in 3 cpp files. Another changes (yml) I tried to compile for old version glibc (for game hosting support)

All works but I can't test it with old linux (old glibc as released in amxmodx website)

File: amxmodx/natives.cpp [hunk 1]
```diff
@@ -112,7 +121,30 @@ int amxx_DynaCallback(int idx, AMX *amx, cell *params)
...
+	using std::chrono::milliseconds;
+
+	auto t1 = high_resolution_clock::now();
+
+	err = amx_Exec(pNative->amx, &ret, pNative->func);
+
+	auto t2 = high_resolution_clock::now();
+
+	auto ms_int = duration_cast<milliseconds>(t2 - t1);
+	auto ms_integer = (int)ms_int.count();
```
Comment (dvander) [inline amxmodx/natives.cpp:141]: This code is repeated almost verbatim four times. It'd be better to factor it out into a helper function and pass the amx_Exec in as a lambda (eg std::function).

File: modules/json/JsonNatives.cpp [hunk 2]
```diff
@@ -697,7 +697,7 @@ static cell AMX_NATIVE_CALL amxx_json_object_set_value(AMX *amx, cell *params)
 		return 0;
 	}
 
-	if (!JsonMngr->IsValidHandle(params[3], Handle_Object))
+	if (!JsonMngr->IsValidHandle(params[3]))
```
Comment (dvander) [inline modules/json/JsonNatives.cpp]: This change appears unrelated.

File: .github/workflows/c-cpp.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    container: s1lentq/linux86buildtools:latest
+
+    steps:
+     - name: Checkout
+       uses: actions/checkout@v2
+       with:
+            fetch-depth: 0
+            submodules: true
+
+     - name: Build using Intel C++ Compiler
```
Comment (dvander) [inline .github/workflows/c-cpp.yml]: There's no need to test ICC.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -134,4 +134,18 @@ jobs:
           cd build
           python3 ../configure.py --enable-optimize --metamod=${{ env.DEPENDENCIES_ROOT_WIN }}\metamod-am --hlsdk=${{ env.DEPENDENCIES_ROOT_WIN }}\hlsdk --mysql=${{ env.DEPENDENCIES_ROOT_WIN }}\mysql-5.5
           ambuild
-          
+      - name: Tar files
```
Comment (dvander) [inline .github/workflows/ci.yml]: All of these CI changes need to be a separate PR. That said, we must only publish artifacts that are created (1) by buildbot and (2) contain code reviewed by a team member. Any artifacts that do not follow these preconditions must be clearly marked as third-party/untrusted builds.

**Comment by dvander** [comment]
I'm not sure what your glibc troubles are, but buildbot uses Debian 7 + clang-3.8 to build AMX Mod X.

File: amxmodx/natives.cpp (hunk 1 above)
Comment (UnrealKaraulov) [reply amxmodx/natives.cpp:141]: yes

**Comment by UnrealKaraulov** [comment]
@dvander I do merge all commits what I can do

I don't know how to make it easy, I tried move performance monitor to amx_Exec function, but got problems with includes like log/pluginmanager etc.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (UnrealKaraulov) [reply .github/workflows/ci.yml]: removed

File: .github/workflows/c-cpp.yml (hunk 3 above)
Comment (UnrealKaraulov) [reply .github/workflows/c-cpp.yml]: removed

File: modules/json/JsonNatives.cpp (hunk 2 above)
Comment (UnrealKaraulov) [reply modules/json/JsonNatives.cpp]: Removed, it just fix error with amxx for my purposes)

File: amxmodx/natives.cpp (hunk 1 above)
Comment (UnrealKaraulov) [reply amxmodx/natives.cpp:141]: But i can't do it )

**Comment by UnrealKaraulov** [comment]
@dvander example output of highload functions (it normally for plugin_init)

L 12/01/2021 - 10:03:28: [admincmd.amxx] performance issue 2. Function plugin_init executed more than 3 ms.
L 12/01/2021 - 10:03:28: [cmdmenu.amxx] performance issue 2. Function plugin_init executed more than 10 ms.
L 12/01/2021 - 10:03:28: [plmenu.amxx] performance issue 2. Function plugin_init executed more than 2 ms.
L 12/01/2021 - 10:03:28: [mapsmenu.amxx] performance issue 2. Function plugin_init executed more than 3 ms.
L 12/01/2021 - 10:03:28: [pluginmenu.amxx] performance issue 2. Function plugin_init executed more than 2 ms.
L 12/01/2021 - 10:03:28: [adminchat.amxx] performance issue 2. Function plugin_init executed more than 2 ms.

**Comment by UnrealKaraulov** [comment]
Closed for update!

**Comment by UnrealKaraulov** [comment]
Code rewrite and updated in new pr: https://github.com/alliedmodders/amxmodx/pull/1019
